frontend/src/visualization/shared-ops/fetchDataFromService.tsx

```diff
@@ -0,0 +1,191 @@
+import Backdrop from '@material-ui/core/Backdrop';
+import CircularProgress from '@material-ui/core/CircularProgress';
+import Button from '@material-ui/core/Button';
+import CloseIcon from '@material-ui/icons/Close';
+import React, { useEffect, useRef, useState } from 'react';
+import { createStyles, makeStyles, Theme } from '@material-ui/core/styles';
+import {
+  CancellationToken,
+  CancellationTokenSource,
+} from '../../utils/CancellationToken';
+
+const useStyles = makeStyles((theme: Theme) =>
+  createStyles({
+    backdrop: {
+      zIndex: theme.zIndex.drawer + 1,
+      color: '#fff',
+    },
+    backdropContent: {
+      marginTop: theme.spacing(3),
+      display: 'flex',
+      flexDirection: 'column',
+      alignItems: 'center',
+    },
+    backdropCancel: {
+      marginTop: theme.spacing(3),
+    },
+    contentContainer: {
+      padding: theme.spacing(3),
+    },
+  })
+);
+
+type RenderFunction<T> = (t: T, update: () => void) => JSX.Element;
+
+type QueryFunction<
+  TArgs extends [...Array<unknown>, CancellationToken],
+  TResult
+> = (...args: TArgs) => Promise<TResult>;
+
+export interface FetchDataOptions<TArgs extends Array<unknown>, TData> {
+  queryFn: QueryFunction<[...TArgs, CancellationToken], TData>;
+  defaultData?: TData;
+}
+
+/**
+ * Fetches data from service using {@link useAsync} with {@link promiseFn} executeQuery.
+ * If the screen is still loading a {@link JSX.Element}, showing a loading screen, is returned.
+ * if an error occurs a {@link JSX.Element}, showing an error screen, is returned.
+ * If the data to be fetched is undefined a {@link JSX.Element}, showing an error message, is returned.
+ *
+ * @param queryFn - the query the service executes
+ * @param service - the service that executes the query
+ * @param arg - optional, additional argument given to {@link useAsync}
+ * @returns if one of the above cases occured, the corresponding {@link JSX.Element}, otherwise the data to be fetched
```

Comment:
Fit documentation to the new implementation.

Reply:
Done!